src/Parallel/ResourceInfo.hpp

```diff
@@ -426,6 +496,12 @@ struct ResourceInfo {
   /// `allocate_array` function of the array component
   const std::unordered_set<size_t>& procs_to_ignore() const;
 
+  /// Returns a `std::set<size_t>` that has all processors with elements on them
+  const std::set<size_t>& procs_with_elements() const;
```

Comment:
still think this needs a better name, maybe `procs_available_for_elements` as there is no guarantee a proc has an element.